[antlr4:x64-windows] Link failure with a freshly installed library

**Describe the bug**
While trying to link to a freshly installed antlr4 C++ runtime, a linker error occurs. The bug occurred on two machines, one with VS 2022, one with VS 2019.

**Environment**
- OS: Windows 10 64-bit, locale pl-PL, used vcpkg before with no issues.
- Compiler: MSVC, Visual Studio Enterprise 2022, 17.4.4 (the bug also occurred in VS Community 2019)

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install antlr4:x64-windows
2. Create a simple project consisting of lexer and parser grammars (can be default grammars from [here](http://lab.antlr.org/)). Contents doesn't matter as long as they're valid. They'll be referred to as ExprLexer and ExprParser respectively.
3. Add a dummy `main` code.
4. Add CMake calls to antlr4 for lexer and parser file generation.
5. Build.
Repro code when
Final CMake file code:
```cmake
cmake_minimum_required(VERSION 3.16)

project(ExprRepro)

find_package(antlr4-runtime CONFIG REQUIRED)
find_package(antlr4-generator CONFIG REQUIRED)

antlr4_generate(
	ExprLexer
	${PROJECT_SOURCE_DIR}/antlr4/ExprLexer.g4
	LEXER
	FALSE
	FALSE
	"Expr"
)
antlr4_generate(
	ExprParser
	${PROJECT_SOURCE_DIR}/antlr4/ExprParser.g4
	PARSER
	FALSE
	FALSE
	"Expr"
	"${ANTLR4_TOKEN_FILES_ExprLexer}"
	"${ANTLR4_TOKEN_DIRECTORY_ExprLexer}"
)

add_executable(ExprMain "Main.cpp") # Main.cpp is just an empty main function

target_include_directories(ExprMain PRIVATE "${ANTLR4_INCLUDE_DIR_ExprLexer}" "${ANTLR4_INCLUDE_DIR_ExprParser}")
target_sources(ExprMain PRIVATE ${ANTLR4_SRC_FILES_ExprLexer} ${ANTLR4_SRC_FILES_ExprParser})
target_include_directories(ExprMain PRIVATE "${ANTLR4_INCLUDE_DIR}")
add_dependencies(ExprMain antlr4_shared)
target_link_libraries(ExprMain PRIVATE antlr4_shared)
target_compile_features(ExprMain PUBLIC cxx_std_17)

```

**Expected behavior**
The code should compile.

**Failure logs**
```
>------ Build started: Project: CMakeLists, Configuration: Debug ------
  [1/6] Generating antlr4_generated_src/ExprLexer/ExprLexer.tokens, antlr4_generated_src/ExprLexer/ExprLexer.interp, antlr4_generated_src/ExprLexer/ExprLexer.h, antlr4_generated_src/ExprLexer/ExprLexer.cpp
  [2/6] Generating antlr4_generated_src/ExprParser/ExprParser.tokens, antlr4_generated_src/ExprParser/ExprParser.interp, antlr4_generated_src/ExprParser/ExprParser.h, antlr4_generated_src/ExprParser/ExprParser.cpp
  [3/6] Building CXX object CMakeFiles\ExprMain.dir\Main.cpp.obj
  [4/6] Building CXX object CMakeFiles\ExprMain.dir\antlr4_generated_src\ExprLexer\ExprLexer.cpp.obj
  [5/6] Building CXX object CMakeFiles\ExprMain.dir\antlr4_generated_src\ExprParser\ExprParser.cpp.obj
  [6/6] Linking CXX executable ExprMain.exe
  FAILED: ExprMain.exe 
  cmd.exe /C "cd . && "E:\Programy\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E vs_link_exe --intdir=CMakeFiles\ExprMain.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- "E:\Programy\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.34.31933\bin\Hostx64\x64\link.exe" /nologo CMakeFiles\ExprMain.dir\Main.cpp.obj CMakeFiles\ExprMain.dir\antlr4_generated_src\ExprLexer\ExprLexer.cpp.obj CMakeFiles\ExprMain.dir\antlr4_generated_src\ExprParser\ExprParser.cpp.obj  /out:ExprMain.exe /implib:ExprMain.lib /pdb:ExprMain.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:console  D:\vcpkg\installed\x64-windows\debug\lib\antlr4-runtime.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cmd.exe /C "cd /D "D:\TheReclif\Documents\C++\GSIL Standalone\out\build\x64-Debug" && C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -noprofile -executionpolicy Bypass -file D:/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary "D:/TheReclif/Documents/C++/GSIL Standalone/out/build/x64-Debug/ExprMain.exe" -installedDir D:/vcpkg/installed/x64-windows/debug/bin -OutVariable out""
  LINK Pass 1: command "E:\Programy\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.34.31933\bin\Hostx64\x64\link.exe /nologo CMakeFiles\ExprMain.dir\Main.cpp.obj CMakeFiles\ExprMain.dir\antlr4_generated_src\ExprLexer\ExprLexer.cpp.obj CMakeFiles\ExprMain.dir\antlr4_generated_src\ExprParser\ExprParser.cpp.obj /out:ExprMain.exe /implib:ExprMain.lib /pdb:ExprMain.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:console D:\vcpkg\installed\x64-windows\debug\lib\antlr4-runtime.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\ExprMain.dir/intermediate.manifest CMakeFiles\ExprMain.dir/manifest.res" failed (exit code 1120) with the following output:
     Creating library ExprMain.lib and object ExprMain.exp
D:\TheReclif\Documents\C++\GSIL Standalone\out\build\x64-Debug\Main.cpp.obj : error LNK2019: unresolved external symbol "const antlr4::atn::SetTransition::`vftable'" (??_7SetTransition@atn@antlr4@@6B@) referenced in function "public: __cdecl antlr4::atn::SetTransition::SetTransition(class antlr4::atn::ATNState *,class antlr4::misc::IntervalSet)" (??0SetTransition@atn@antlr4@@QEAA@PEAVATNState@12@VIntervalSet@misc@2@@Z)
D:\TheReclif\Documents\C++\GSIL Standalone\out\build\x64-Debug\ExprLexer.cpp.obj : error LNK2001: unresolved external symbol "const antlr4::atn::SetTransition::`vftable'" (??_7SetTransition@atn@antlr4@@6B@)
D:\TheReclif\Documents\C++\GSIL Standalone\out\build\x64-Debug\ExprParser.cpp.obj : error LNK2001: unresolved external symbol "const antlr4::atn::SetTransition::`vftable'" (??_7SetTransition@atn@antlr4@@6B@)
D:\TheReclif\Documents\C++\GSIL Standalone\out\build\x64-Debug\ExprMain.exe : fatal error LNK1120: 1 unresolved externals
  
  ninja: build stopped: subcommand failed.

Build failed.
```

**Additional context**
**It's possible to build the project with manually built libraries** (i.e. directly from antlr4 Github repo) using `target_link_libraries(ExprMain PRIVATE "${PATH_TO_ANTLR4_LIB_DIR}/dist/antlr4-runtime.lib")` so I assume the issue lays somewhere on the port side.

The unresolved symbol `??_7SetTransition@atn@antlr4@@6B@` can be found both in the DLL and static library that's used to link to it using `dumpbin /EXPORTS`.

The bug persists in both Debug and Release configurations.

**EDIT** Bug description

--- a/ports/antlr4/portfile.cmake
+++ b/ports/antlr4/portfile.cmake
@@ -8,7 +8,7 @@ vcpkg_from_github(
     SHA512 a52356410c95ec6d7128b856dcf4c20a17cdd041270d2c4d700ef02ea715c87a00a87c2ad560277424b300435c6e9b196c8bc9c9f50ae5b6804d8214b4d397d0
     PATCHES
         fix_build_4.11.1.patch
-		set-export-macro-define-as-private.patch
+        set-export-macro-define-as-private.patch
 )
 
 set(RUNTIME_PATH "${SOURCE_PATH}/runtime/Cpp")

--- a/ports/antlr4/portfile.cmake
+++ b/ports/antlr4/portfile.cmake
@@ -8,6 +8,7 @@ vcpkg_from_github(
     SHA512 a52356410c95ec6d7128b856dcf4c20a17cdd041270d2c4d700ef02ea715c87a00a87c2ad560277424b300435c6e9b196c8bc9c9f50ae5b6804d8214b4d397d0
     PATCHES
         fix_build_4.11.1.patch
+		set-export-macro-define-as-private.patch
 )
 
 set(RUNTIME_PATH "${SOURCE_PATH}/runtime/Cpp")

--- a/ports/antlr4/set-export-macro-define-as-private.patch
+++ b/ports/antlr4/set-export-macro-define-as-private.patch
@@ -1,0 +1,13 @@
+diff --git a/runtime/Cpp/runtime/CMakeLists.txt b/runtime/Cpp/runtime/CMakeLists.txt
+index 19cc21c..2a33fe9 100644
+--- a/runtime/Cpp/runtime/CMakeLists.txt
++++ b/runtime/Cpp/runtime/CMakeLists.txt
+@@ -103,7 +103,7 @@ set(static_lib_suffix "")
+ if (WIN32)
+   set(static_lib_suffix "-static")
+   if(BUILD_SHARED_LIBS)
+-  target_compile_definitions(antlr4_shared PUBLIC ANTLR4CPP_EXPORTS)
++  target_compile_definitions(antlr4_shared PRIVATE ANTLR4CPP_EXPORTS)
+   else()
+   target_compile_definitions(antlr4_static PUBLIC ANTLR4CPP_STATIC)
+   endif()

--- a/ports/antlr4/vcpkg.json
+++ b/ports/antlr4/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "antlr4",
   "version": "4.11.1",
+  "port-version": 1, 
   "description": "ANother Tool for Language Recognition",
   "homepage": "https://www.antlr.org",
   "license": "BSD-3-Clause",

--- a/ports/antlr4/vcpkg.json
+++ b/ports/antlr4/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "antlr4",
   "version": "4.11.1",
-  "port-version": 1, 
+  "port-version": 1,
   "description": "ANother Tool for Language Recognition",
   "homepage": "https://www.antlr.org",
   "license": "BSD-3-Clause",

--- a/versions/a-/antlr4.json
+++ b/versions/a-/antlr4.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "162afe92eab004d57c6b0c972ea5e9b371ccb23a",
+      "version": "4.11.1",
+      "port-version": 1
+    },
+    {
       "git-tree": "adff4448fb67cdc7d7a478d5f5f3e973ad65a386",
       "version": "4.11.1",
       "port-version": 0

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -114,7 +114,7 @@
     },
     "antlr4": {
       "baseline": "4.11.1",
-      "port-version": 0
+      "port-version": 1
     },
     "any-lite": {
       "baseline": "0.4.0",

